ldap_auth_provider.py

```diff
@@ -347,6 +384,16 @@ class _LdapConfig(object):
             "mail",
         ])
 
+        ldap_config.active_directory = config.get("active_directory", False)
+        if ldap_config.active_directory:
+            if config["attributes"]["uid"].lower() != "userprincipalname":
+                raise Exception(
+                    "Only userPrincipalName is supported "
+                    "as uid property in active_directory mode"
+                )
```

Comment:
ok, but it seems like this ought to be handled via documentation, rather than a hardcoded check. people may have systems that behave similarly to AD but expose a UPN-like attribute with a different name: there's no reason not to support that.